frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -99,6 +87,23 @@ func newCVDExecContext(execContext ExecContext, usr *user.User) cvd.CVDExecConte
 	}
 }
 
+// Creates a CVD execution context from a regular execution context
+// for a given user. If user is nil, the running process's user will be used.
+func newCVDExecContextWithUser(execContext ExecContext, usr *user.User) cvd.CVDExecContext {
+	if usr == nil {
+		return newCVDExecContext(execContext)
+	}
```

Comment:
This defeats the purpose of the refactor. Either return an error on `nil` or make the parameter not a pointer.